### PR TITLE
Addition of sbt-idea plugin.

### PR DESCRIPTION
**Pull request by kermitas**
Hi,
I just added sbt-idea plugin. Also refactor .gitignore and plugins.sbt to make them easier to read.

### CODE REVIEW

**Comment by gkleiman** [comment]
Thank you for contributing to Marathon!

The name of the commit related to .gitignore is misleading, as new entries are added as well.

It would also be good if you could follow our [Contributor Guidelines](https://mesosphere.github.io/marathon/docs/contributing.html) and either squash the commits or create one PR per commit.

I'll let others comment on the idea of adding the sbt-idea plugin. I personally use IntelliJ IDEA, but creating a project from existing sources has been always enough for me.

**Comment by kermitas** [comment]
"The name of the commit related to .gitignore is misleading, as new entries are added as well." - that is right, sorry for that (few new entries were added, for ignoring Idea and Eclipse files)

"It would also be good if you could follow our Contributor Guidelines and either squash the commits or create one PR per commit." - ok, thanks for the hint, I will

"I'll let others comment on the idea of adding the sbt-idea plugin. I personally use IntelliJ IDEA, but creating a project from existing sources has been always enough for me." - for me it was crappy for a years so I am using sbt-idea; I have no idea how well Idea reads SBT's configuration (that one can put into series of *.scala and *.sbt files) but sbt-idea gives you simple `gen-idea` command that generates up to date project (with whatever was in SBT configuration)

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
-docker-volumes
-target
-.reviewboardrc
 dependency-reduced-pom.xml
 node_modules
+.Rproj.user
+start.sh
+
+# === Review Board (https://www.reviewboard.org/) specific files
+.reviewboardrc
```
Comment (aquamatthias) [inline .gitignore]: We don't use review board

**Comment by aquamatthias** [comment]
Hey @kermitas thanks for this PR. I added some suggestions. Please address and I will merge.

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,29 +1,42 @@
+// This file contains a list of SBT plugins.
+
+// === resolvers
 resolvers += Classpaths.typesafeReleases
+resolvers += Classpaths.sbtPluginReleases
+resolvers += Resolver.sonatypeRepo("releases")
+resolvers += Resolver.sonatypeRepo("snapshots")
```
Comment (aquamatthias) [inline project/plugins.sbt]: We try to use only stable dependencies

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+resolvers += "Era7 maven releases" at "http://releases.era7.com.s3.amazonaws.com"
 
-resolvers += "sonatype-releases" at "https://oss.sonatype.org/content/repositories/releases/"
+// === dependecy graph, https://github.com/jrudolph/sbt-dependency-graph (e.g. `dependencyGraphMl` command)
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.7.5")
 
+// === IntelliJ Idea project generator, https://github.com/mpeltonen/sbt-idea (e.g. `gen-idea` command)
+addSbtPlugin("com.github.mpeltonen" % "sbt-idea" % "1.7.0-SNAPSHOT")
```
Comment (aquamatthias) [inline project/plugins.sbt]: Such plugins are best kept as sbt global dependencies

File: .gitignore (hunk 1 above)
Comment (kermitas) [inline .gitignore]: .reviewboardrc was here before my changes; if it is not needed we can remove it

File: project/plugins.sbt (hunk 2 above)
Comment (kermitas) [inline project/plugins.sbt]: Even for SBT plugins (that affects only SBT)?

File: project/plugins.sbt (hunk 3 above)
Comment (kermitas) [inline project/plugins.sbt]: Great, thanks for hint!
http://www.scala-sbt.org/0.13/tutorial/Using-Plugins.html#Global+plugins

File: project/plugins.sbt (hunk 2 above)
Comment (aquamatthias) [inline project/plugins.sbt]: To have reproducible builds, I would say so.

Comment (kermitas) [inline project/plugins.sbt]: Ok.

File: .gitignore (hunk 1 above)
Comment (aquamatthias) [inline .gitignore]: I was not aware - thanks.

**Comment by aquamatthias** [comment]
Close this, since the original purpose seem obsolete.

**Pull request closed without merging**